components/dot_menu/index.ts

```diff
@@ -54,6 +61,9 @@ function mapStateToProps(state: GlobalState, ownProps: Props) {
     const currentTeam = getCurrentTeam(state) || {};
     const currentTeamUrl = `${getSiteURL()}/${currentTeam.name}`;
 
+    const apps = appsEnabled(state);
+    const appBindings = apps ? getAppBindings(state, AppBindingLocations.POST_MENU_ITEM) : [];
```

Comment:
(6/5) Here's another place where we're passing a new empty array into the component which is going to cause a lot of unnecessary rerendering.

That said, as mentioned in https://github.com/mattermost/mattermost-redux/pull/1414#discussion_r596131198, the selector is also going to cause a lot of performance issues as well since it can return a new array every time its called. I'd suggest making a selector factory and using it once for each location. And unless we care about getting app bindings when apps are disabled, I'd suggest making that return an empty array itself when apps are disabled instead of having to do that everywhere the selector is used. Something like this would work great

```javascript
// This is slightly different than other selector factories since we're binding location at creation time since it's an enum
const makeGetAppBindings = (location) => {
    return createSelector(
        appsEnabled,
        (state) => state.entities.apps.bindings,
        (enabled, allBindings) => {
            if (!enabled || !state.entities.apps.bindings) {
            return [];
        }
        
        return /* filter and reduce as necessary */;
    },
};

const getAllAppBindings = makeGetAppBindings('');

// Since each location is only used by a single component from what I can tell, these can just be defined in the component's index file outside of mapStateToProps
const getPostMenuAppBindings = makeGetPostMenuAppBindings(AppBindingLocations.POST_MENU_ITEM);
```

Reply:
Selector factory created and used for all locations now